addons/account/models/account_account.py

```diff
@@ -188,6 +188,32 @@ def _check_user_type_id_sales_purchase_journal(self):
         if self._cr.fetchone():
             raise ValidationError(_("The account is already in use in a 'sale' or 'purchase' journal. This means that the account's type couldn't be 'receivable' or 'payable'."))
 
+    @api.constrains('reconcile')
+    def _check_used_as_journal_default_debit_credit_account(self):
+        if not self:
+            return
+
+        self.flush(['reconcile'])
+        accounts = self.filtered(lambda a: not a.reconcile)
+        if accounts:
+            self._cr.execute('''
+                SELECT journal.name, journal.id
+                FROM account_journal journal
+                WHERE journal.payment_credit_account_id in %(credit_account)s
+                OR journal.payment_debit_account_id in %(debit_account)s ;
+            ''', {
+                'credit_account': tuple(accounts.ids),
+                'debit_account': tuple(accounts.ids)
+            })
+
+            records = self._cr.fetchall()
```

Comment:
`rows` because I `records` == recorset of journals

Reply:
The convention is `SELECT journal_id`, then `journals = self.env['account.journal'].browse(...)` in order to use the right display_name: `journals.mapped('display_name')`

Reply:
Done!